test_conformance/api/test_platform.cpp

```diff
@@ -17,131 +17,85 @@
 
 #include <string.h>
 
-#define EXTENSION_NAME_BUF_SIZE 4096
-
-#define PRINT_EXTENSION_INFO 0
+#include <algorithm>
 
 int test_platform_extensions(cl_device_id deviceID, cl_context context,
                              cl_command_queue queue, int num_elements)
 {
-    const char *extensions[] = { "cl_khr_byte_addressable_store",
-                                 //    "cl_APPLE_SetMemObjectDestructor",
-                                 "cl_khr_global_int32_base_atomics",
-                                 "cl_khr_global_int32_extended_atomics",
-                                 "cl_khr_local_int32_base_atomics",
-                                 "cl_khr_local_int32_extended_atomics",
-                                 "cl_khr_int64_base_atomics",
-                                 "cl_khr_int64_extended_atomics",
-                                 // need to put in entires for various atomics
-                                 "cl_khr_3d_image_writes", "cl_khr_fp16",
-                                 "cl_khr_fp64", NULL };
-
-    bool extensionsSupported[] = {
-        false, //"cl_khr_byte_addressable_store",
-        false, // need to put in entires for various atomics
-        false, // "cl_khr_global_int32_base_atomics",
-        false, // "cl_khr_global_int32_extended_atomics",
-        false, // "cl_khr_local_int32_base_atomics",
-        false, // "cl_khr_local_int32_extended_atomics",
-        false, // "cl_khr_int64_base_atomics",
-        false, // "cl_khr_int64_extended_atomics",
-        false, //"cl_khr_3d_image_writes",
-        false, //"cl_khr_fp16",
-        false, //"cl_khr_fp64",
-        false // NULL
-    };
-
-    int extensionIndex;
-
     cl_platform_id platformID;
     cl_int err;
 
-    char platform_extensions[EXTENSION_NAME_BUF_SIZE];
-    char device_extensions[EXTENSION_NAME_BUF_SIZE];
-
-    // Okay, so what we're going to do is just check the device indicated by
-    // deviceID against the platform that includes this device
+    err = clGetDeviceInfo(deviceID, CL_DEVICE_PLATFORM, sizeof(cl_platform_id),
+                          (void *)(&platformID), NULL);
+    test_error(err, "clGetDeviceInfo failed for CL_DEVICE_PLATFORM");
 
+    // Note that we read into a vector for tokenization, rather than a string,
+    // because strtok_r modifies the string.
 
-    // pass CL_DEVICE_PLATFORM to clGetDeviceInfo
-    // to get a result of type cl_platform_id
+    size_t size = 0;
 
-    err = clGetDeviceInfo(deviceID, CL_DEVICE_PLATFORM, sizeof(cl_platform_id),
-                          (void *)(&platformID), NULL);
+    err = clGetPlatformInfo(platformID, CL_PLATFORM_EXTENSIONS, 0, NULL, &size);
+    test_error(err,
+               "clGetPlatformInfo failed to get CL_PLATFORM_EXTENSIONS size");
 
-    if (err != CL_SUCCESS)
-    {
-        vlog_error("test_platform_extensions : could not get platformID from "
-                   "device\n");
-        return -1;
-    }
+    std::vector<char> platformExtensions(size);
```

Comment:
I think this would be better served with a `std::string`, which would then unlock `find` and `substr` in the loop below.